src/components/Mission.js

```diff
@@ -1,13 +1,41 @@
 import React from 'react';
 
 const Mission = (mission) => {
-  const { id, name, description } = mission;
+  const {
+    id,
+    name,
+    description,
+    joinMission,
+    leaveMission,
+    joined,
+  } = mission;
+
+  const handleSubmit = (e) => {
+    if (!joined) {
+      e.target.classList.remove('leave');
+      e.target.classList.add('joined');
+      joinMission(id);
+    } else {
+      e.target.classList.remove('joined');
+      e.target.classList.add('leave');
+      leaveMission(id);
+    }
+  };
+
   return (
     <tr key={id}>
       <td>{name}</td>
       <td>{description}</td>
-      <td><p>NOT A MEMBER</p></td>
-      <td><button className="btn" type="button">Join Mission</button></td>
+      <td><p className="member">NOT A MEMBER</p></td>
+      <td>
+        <button
+          className="btn"
+          onClick={handleSubmit}
+          type="button"
+        >
+          {joined ? 'Leave Mission' : 'Join Mission'}
+        </button>
+      </td>
```

Comment:
The "Leave Mission" button style changes back to black (default style) when you switch to another page and then go back to the missions page.
![image](https://user-images.githubusercontent.com/20567503/161858849-891d84f4-ef82-4b34-b0c4-94a92ecb8e50.png)
I suggest you look into that, maybe add another condition for the style itself.

Reply:
Thanks for alerting me on this. I will try and resolve that.